nav2_costmap_2d/plugins/voxel_layer.cpp

```diff
@@ -334,9 +335,30 @@ void VoxelLayer::raytraceFreespace(
   }
 
   if (publish_clearing_points) {
-    clearing_endpoints_->points.clear();
-    clearing_endpoints_->points.reserve(clearing_observation_cloud_size);
+    clearing_endpoints_->data.clear();
+    clearing_endpoints_->width = clearing_observation.cloud_->width;
+    clearing_endpoints_->height = clearing_observation.cloud_->height;
+    clearing_endpoints_->is_dense = true;  // are there no invalid points in the cloud ?
+    clearing_endpoints_->is_bigendian = false;
+  }else{
+    clearing_endpoints_->width  = 0;
+    clearing_endpoints_->height = 0;
   }
+  // FIXME(sachin): Should I use clearing_observation.cloud_->fields.size() ??
```

Comment:
@saching13 please remove all TODOs or FIXME's in the code, we don't allow for those. If there's a question, ask us please. If it can be removed, remove it